### PR TITLE
Feature/marp 1018 new connector rtf factory

### PR DESCRIPTION
**Pull request by tvtphuc-axonivy**

### CODE REVIEW

File: rtf-factory/src/ch/ivyteam/ivy/RtfFactory/RtfExpander.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    String str = new String(Files.readAllBytes(file.toPath()));
+    String expandedStr = expand(str, in);
+
+    FacesContext facesContext = FacesContext.getCurrentInstance();
+    HttpServletResponse response = (HttpServletResponse) facesContext.getExternalContext().getResponse();
+    response.reset();
+    response.setHeader("Content-Type","txt/rtf");
+    response.addHeader("Content-Disposition","attachment; filename="+file.getName());
```
Comment (tvtphuc-axonivy) [inline rtf-factory/src/ch/ivyteam/ivy/RtfFactory/RtfExpander.java:37]: need to create a constant for all hard-coded string.

**Review by tvtphuc-axonivy — COMMENTED** [review]
Hi @ivy-bb , I have reset the master branch to your initial commit. I cloned your code to the branch "feature/MARP-1018-New-connector-RTF-Factory"
and then created a PR to the master branch that has some feedback needed to be addressed.

**Comment by tvtphuc-axonivy** [comment]
Hi @ivy-bb I did an enhancement and format for this connector.
Could you help to review and give your feedback? then I can continue for onboarding it. Thanks!
FYI @ivy-sgi

File: rtf-factory/src/ch/ivyteam/ivy/RtfFactory/ExportFromCms.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import ch.ivyteam.ivy.environment.Ivy;
+import ch.ivyteam.ivy.scripting.objects.File;
+
+public class ExportFromCms {
+
+  public static java.io.File export(String cmsUri, String ext) throws IOException {
```
[the remark this answers is not in the document]
Comment (ivy-bb) [reply rtf-factory/src/ch/ivyteam/ivy/RtfFactory/ExportFromCms.java]: I have improved the included cms text.

**Comment by ivy-bb** [comment]
Thank you very much for your good work. 
I finally found the time to review.  For me it looks very smart now.

**Pull request merged**